src/main/java/com/wanderersoftherift/wotr/block/TrapBlock.java

```diff
@@ -38,13 +41,14 @@ public TrapBlock(Properties properties) {
         this.registerDefaultState(this.defaultBlockState().setValue(WAITING_FOR_TICK, false));
     }
 
-    protected void tick(@NotNull BlockState state, @NotNull ServerLevel level, @NotNull BlockPos pos, @NotNull RandomSource random) {
+    @Override
+    protected void tick(BlockState state, ServerLevel level, BlockPos pos, RandomSource random) {
         if (state.getValue(DEACTIVATED)) {
             level.setBlockAndUpdate(pos, state.setValue(DEACTIVATED, false).setValue(STAGE, 0).setValue(WAITING_FOR_TICK, false));
             return;
         }
 
-        if (state.getValue(STAGE) + 1 > STAGES) {
+        if (state.getValue(STAGE) == STAGES) {
```

Comment:
I would suggest >= here just in case the value slops past (perhaps because of a code change while the game is saved that reduces the number of stages).